adi/ad7606.py

```diff
@@ -103,24 +117,32 @@ def scale(self, value):
             self._set_iio_attr(self.name, "scale", False, str(Decimal(value).real))
 
         @property
-        def range(self):
-            """AD7606 channel range"""
-            return self._get_iio_attr(self.name, "range", False)
-
-        @range.setter
-        def range(self, value):
-            self._set_iio_attr(self.name, "range", False, str(Decimal(value).real))
+        def scale_available(self):
+            """'scale_available' is available per channel on AD7606C-16 and AD7606C-18 in SW mode"""
+            # We can't detect SW mode, so we use a try block
+            try:
+                return self._get_iio_attr_str(self.name, "scale_available", False)
+            except:
+                return self._dev.scale_available
 
     def to_volts(self, index, val):
         """Converts raw value to SI"""
         _scale = self.channel[index].scale
 
-        ret = None
+        if isinstance(val, int):
+            return val * _scale
+
+        if isinstance(val, list):
+            return [x * _scale for x in val]
+
+        # ADC7606C-18 will return int32 samples from the driver
+        if isinstance(val, np.int32):
+            return val * _scale
 
         if isinstance(val, np.int16):
-            ret = val * _scale
+            return val * _scale
 
         if isinstance(val, np.ndarray):
-            ret = [x * _scale for x in val]
+            return [x * _scale for x in val]
 
-        return ret
+        return None
```

Comment:
If replacing with returns you can omit last line. Python always returns None if there is no explicit return statement